pandas/tests/arithmetic/test_timedelta64.py

```diff
@@ -31,12 +44,92 @@
     Int64Index,
     UInt64Index,
 )
+from pandas.core.arrays import (
+    DatetimeArray,
+    TimedeltaArray,
+)
 from pandas.tests.arithmetic.common import (
     assert_invalid_addsub_type,
     assert_invalid_comparison,
     get_upcast_box,
 )
 
+timedelta_types = (Timedelta, TimedeltaArray, TimedeltaIndex, Series, DataFrame)
+timestamp_types = (Timestamp, DatetimeArray, DatetimeIndex, Series, DataFrame)
+containers = slice(1, None)
```

Comment:
ok can you move all of these changes (and the existing overflow tests) to a new file, test_timedelta64_overflow.py or similar. We use hypothesis in very specific ways and most of this is just test setup which is  confusing.

Reply:
> ok can you move all of these changes (and the existing overflow tests) to a new file

@jreback sure thing 👍. I've moved ([bd62e08](https://github.com/pandas-dev/pandas/pull/46854/commits/bd62e0844af1604089c7bd7e7a4bb06f0ddaf5af)) all of the new overflow tests, plus those that already existed in the 2 files this PR touches, to `pandas/tests/test_timedelta64_overflow.py`.

Some td64 overflow tests remain in other modules. Should I also move these, or should they stay in place?   
  - `tests/tslibs/test_conversion.py::test_ensure_timedelta64ns_overflows()`
  - `tests/tslibs/test_timedeltas.py::test_huge_nanoseconds_overflow()`
  - `tests/scalar/timedelta/test_timedelta.py::test_mul_preserves_reso()`
  - `tests/scalar/timedelta/test_constructors.py::test_construct_from_td64_with_unit(),test_overflow_on_construction()`

Happy to do either, just wondering what the preferred organization is? I left the above stuff in place based on the recommendation 1 in the [docs](https://github.com/pandas-dev/pandas/blob/8980af7ce9d98713b0f8792e38f0fe43088e8780/doc/source/development/test_writing.rst).


> We use hypothesis in very specific ways and most of this is just test setup which is confusing.

Yeah, the new tests probably use it gratuitously 😬 I'll pare that back and try to better match the existing usage patterns.

Reply:
> Some td64 overflow tests remain in other modules. Should I also move these, or should they stay in place?

these should stay in place